services/motion/builtin/move_request.go

```diff
@@ -179,6 +222,123 @@ func (mr *moveRequest) deviatedFromPlan(ctx context.Context, waypoints state.Way
 	return state.ExecuteResponse{}, nil
 }
 
+// want func that returns obstacles in worldstate.
+func (mr *moveRequest) getTransientDetections(
+	ctx context.Context,
+	visSrvc vision.Service,
+	camName resource.Name,
+	localizerCurrentPosition *referenceframe.PoseInFrame,
+) ([]*referenceframe.GeometriesInFrame, []*referenceframe.GeometriesInFrame, error) {
+	mr.logger.CDebugf(ctx,
+		"proceeding to get detections from vision service: %s with camera: %s",
+		visSrvc.Name().ShortName(),
+		camName.ShortName(),
+	)
+	// Any obstacles specified by the worldstate of the moveRequest will also re-detected here.
+	// There is no need to append the new detections to the existing worldstate.
+	// We can safely build from scratch without excluding any valuable information.
+
+	// get detections from vision service
+	detections, err := visSrvc.GetObjectPointClouds(ctx, camName.Name, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	mr.logger.CDebugf(ctx, "got %d detections", len(detections))
+
+	// Here we make the assumption that the localizer's current position is shared with the observing camera
+	// TODO: ENSURE THIS ASSUMPTION HOLDS
+	currentPositionInWorld, err := mr.fsService.TransformPose(ctx, localizerCurrentPosition, "world", nil)
+	if err != nil {
+		currentPositionInWorld = localizerCurrentPosition
+	}
+	mr.logger.CDebugf(ctx, "currentPositionInWorld: %v", spatialmath.PoseToProtobuf(currentPositionInWorld.Pose()))
+
+	// determine transform of camera to world
+	cameraOrigin := referenceframe.NewPoseInFrame(camName.ShortName(), spatialmath.NewZeroPose())
+	cameraToWorld, err := mr.fsService.TransformPose(ctx, cameraOrigin, "world", nil)
+	if err != nil {
+		// here we make the assumption the camera is coincident with the world
+		mr.logger.CDebugf(ctx,
+			"we assume the world is coincident with the camera named: %s due to err: %v",
+			camName.ShortName(), err.Error(),
+		)
+		cameraToWorld = cameraOrigin
+	}
+	mr.logger.CDebugf(ctx, "cameraToWorld transform: %v", spatialmath.PoseToProtobuf(cameraToWorld.Pose()))
+
+	// detections in the world frame
+	absoluteGeoms := []spatialmath.Geometry{}
+	// detection in the base frame
+	relativeGeoms := []spatialmath.Geometry{}
+	for i, detection := range detections {
+		geometry := detection.Geometry
+		// if more than 1.5 meters away lets not concern ourselves with avoiding this detection
+		if geometry.Pose().Point().Norm() > 1500 {
```

Comment:
Can this 1500 mm please not be hard coded. Can we please move this up to a package level constant?